test/util/TypeTraitsTestHelpers.h

```diff
@@ -16,11 +31,9 @@ parameter type list with itself, as template parameters. For example: If given
 #include <concepts>
 template <typename Func, typename... Parameter>
 constexpr void passCartesianPorductToLambda(Func func) {
-  (
-      [&func]<typename T>() {
-        (func.template operator()<T, Parameter>(), ...);
-      }.template operator()<Parameter>(),
-      ...);
+  (detail::FunctionCaller<Func, Parameter...>{func}
+       .template operator()<Parameter>(),
```

Comment:
Refactor this util to use a `type_identity` as a first argument and adapt the call sites.

Reply:
Addressed